homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,184 @@
+        try:
+            self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        try:
+            afsapi = AFSAPI(self._webfsapi_url, import_info[CONF_PIN])
+
+            self._unique_id = await afsapi.get_radio_id()
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except InvalidPinException:
+            return self.async_abort(reason="invalid_auth")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        await self.async_set_unique_id(self._unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured()
+
+        self._name = import_info[CONF_NAME] or "Radio"
+
+        return await self._create_entry(pin=import_info[CONF_PIN])
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step of manual configuration."""
+        errors = {}
+
+        if user_input:
+            device_url = (
+                f"http://{user_input[CONF_HOST]}:{user_input[CONF_PORT]}/device"
+            )
+            try:
+                self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+            except FSConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception as exception:  # pylint: disable=broad-except
+                _LOGGER.exception(exception)
+                errors["base"] = "unknown"
+            else:
+                return await self._async_step_device_config_if_needed()
+
+        data_schema = self.add_suggested_values_to_schema(
+            STEP_USER_DATA_SCHEMA, user_input
+        )
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def _async_step_device_config_if_needed(self) -> FlowResult:
+        """Most users will not have changed the default PIN on their radio.
+
+        We try to use this default PIN, and only if this fails ask for it via `async_step_device_config`
+        """
+
+        try:
+            # try to login with default pin
+            afsapi = AFSAPI(self._webfsapi_url, DEFAULT_PIN)
+
+            self._name = await afsapi.get_friendly_name()
+
+            self.context["title_placeholders"] = {"name": self._name}
+
+            self._unique_id = await afsapi.get_radio_id()
+            await self.async_set_unique_id(self._unique_id)
+            self._abort_if_unique_id_configured()
+
+            return await self._create_entry()
+        except InvalidPinException:
+            pass  # Ask for a PIN
```

Comment:
I think there is too much inside the `try`
Which lines are actually likely to raise `InvalidPinException`?
Maybe you should refactor this with `try...except...else`